main/webapp/modules/core/scripts/dialogs/add-rows-dialog.js

```diff
@@ -0,0 +1,101 @@
+const AddRowsDialog = (function (path) {
```

Comment:
What is the purpose of declaring this as a function that gets applied immediately? Does it delay the evaluation of some instructions to later?

Reply:
It's an older designed pattern in JavaScript called [immediately invoked function expressions (IIFE)](https://developer.mozilla.org/en-US/docs/Glossary/IIFE). JavaScript modules arguably make it obsolete now, but it actually works well with how scripts are bundled in OpenRefine. It prevents pollution of the global namespace and create private and public variables and methods as in the module pattern.